app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt

```diff
@@ -1167,13 +1163,20 @@ class MainViewModel @Inject constructor(
     //endregion
 
     //region Phishing
-    fun reportPhishing(threadUid: String, message: Message) = viewModelScope.launch(ioCoroutineContext) {
+    fun reportPhishing(threadUids: List<String>, messages: List<Message>) = viewModelScope.launch(ioCoroutineContext) {
         val mailboxUuid = currentMailbox.value?.uuid!!
+        val messagesUids: List<String> = messages.map { it.uid }
 
-        with(ApiRepository.reportPhishing(mailboxUuid, message.folderId, message.shortUid)) {
-
+        with(ApiRepository.reportPhishing(mailboxUuid, messagesUids)) {
```

Comment:
You could check if the messagesUids list is empty before making the call.
If it is empty, you can display "an error has occured" and return